packages/react-effect/src/server.tsx

```diff
@@ -1,19 +1,24 @@
 import * as React from 'react';
 import {renderToStaticMarkup} from 'react-dom/server';
 import {EffectContext, EffectManager} from './context';
+import {Pass} from './types';
 
 interface Options {
   include?: symbol[] | boolean;
+  maxPasses?: number;
   decorate?(element: React.ReactElement<any>): React.ReactElement<any>;
   renderFunction?(element: React.ReactElement<{}>): string;
-  betweenEachPass?(): any;
-  afterEachPass?(): any;
+  betweenEachPass?(pass: Pass): any;
+  afterEachPass?(pass: Pass): any;
 }
 
+const DEFAULT_MAX_PASSES = 5;
```

Comment:
5 feels like a silent footgun for consumers, but you're way more aware of realistic numbers for this so, 🔋

If someone wants to be proactive about this, they could add `afterEachPass` handler that logs/alerts when their app approaches the limit, right?

Reply:
Yes, I will add thorough logging for this on web's side